src/ai-examples/azext_ai_examples/custom.py

```diff
@@ -78,20 +80,20 @@ def ping_aladdin_service():
 
 
 def call_aladdin_service(query):
-    correlation_id = telemetry_core._session.correlation_id  # pylint: disable=protected-access
-    subscription_id = telemetry_core._get_azure_subscription_id()  # pylint: disable=protected-access
     version = str(parse_version(core_version))
 
+    # Only pull in the contextual values if we have consent
+    telemetry_consent = telemetry_core.is_telemetry_enabled()
+    correlation_id = telemetry_core._session.correlation_id if telemetry_consent else DEFAULT_GUID  # pylint: disable=protected-access
```

Comment:
when user chose opt out of telemetry collecting, we should not collect telemetry. then what's the default value used for?

Reply:
Our service expects to receive a GUID value for either `correlationId` or `subscriptionId` if one is supplied. The `DEFAULT_GUID` just acts as a dummy value (being all zeros) to make the service happy. The other option would be to not include `correlationId` or `subscriptionId` in the request at all, but that made the flow a little more complex, so I thought this would be a simpler solution. I'm willing to switch to the other way if you think that would be preferred.

Reply:
thanks @mirdaki . Just to confirm, when user opting out, is there another other data sent to the service besides correlationId and subsId? We follows Microsoft general practice to allow user disabling telemetry. I'm not sure is it right to still send data when user opting out, even some dummy data.

Reply:
That's a good point. Thank you for bringing it up. I will get an update out later today!

Reply:
@yungezz Pushed the update. Let me know what you think!

And to answer your question about what data we send (sorry, forgot to reply before), in addition to the optional `correlationId` and `subscriptionId`, we need the command the user is asking about and the version of the CLI so we can provide the appropriate examples. Those last two items are required for us to give useful examples, so we don't consider them telemetry.

The reason we collect the `correlationId` and `subscriptionId` (for the users who do allow it) is so we can better understand the context users are running commands in to improve our ML model.